### PR TITLE
Fix various flaky tests

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTests.kt [hunk 1]
```diff
@@ -441,27 +441,6 @@ class LoggingTest : ClientLoader() {
         }
     }
 
-    @Test
-    fun testCustomServer() = clientTests(listOf("iOS", "native:CIO")) {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTests.kt:445]: can we replace this with a test server?

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt [hunk 2]
```diff
@@ -92,17 +92,16 @@ internal fun CoroutineScope.attachForReadingDirectImpl(
         channel.writeSuspendSession {
             while (true) {
                 var rc = 0
+                val buffer = request(1)
+                if (buffer == null) {
+                    if (channel.isClosedForWrite) break
```
Comment (rsinukov) [inline ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:97]: I think we should do the same for `isClosedForWrite`. We should not lose exceptions in writing.

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-tests/build.gradle.kts [hunk 3]
```diff
@@ -113,9 +113,13 @@ kotlin.sourceSets {
         } else {
             val posixTest by getting {
                 dependencies {
+                    val hostname: String by project.ext
                     // api(project(":ktor-client:ktor-client-ios"))
                     api(project(":ktor-client:ktor-client-curl"))
-                    api(project(":ktor-client:ktor-client-cio"))
+
+                    if (!hostname.startsWith("win")) {
+                        api(project(":ktor-client:ktor-client-cio"))
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-tests/build.gradle.kts:121]: It's only in IDEA and for native CIO. It's unsupported on Win

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTests.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTests.kt:445]: We have similar tests with our internal server test server

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt (hunk 2 above)
Comment (e5l) [reply ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:97]: Sure, will fix

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt [hunk 4]
```diff
@@ -415,7 +415,7 @@ class HttpTimeoutTest : ClientLoader() {
     }
 
     @Test
-    fun testConnectionRefusedException() = clientTests(listOf("Js", "native:*")) {
+    fun testConnectionRefusedException() = clientTests(listOf("Js", "native:*", "win:*")) {
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt:418]: We need a ticket to not forget about this thing: one day we'll enable it back, right?

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt [hunk 5]
```diff
@@ -92,17 +92,16 @@ internal fun CoroutineScope.attachForReadingDirectImpl(
...
+                    channel.flush()
+                    tryAwait(1)
```
Comment (cy6erGn0m) [inline ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:99]: Why don't we control timeout here anymore?

**Review by cy6erGn0m — APPROVED** [review]

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt (hunk 4 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt:418]: Nope, the original test was muted cause this kind of timeout is not supported on windows

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt (hunk 5 above)
Comment (e5l) [reply ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:99]: We use timeout only for network operations

Comment (e5l) [reply ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:99]: btw it works in the same way in similar places

Comment (rsinukov) [reply ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:99]: yes, previous behavior was wrong

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt [hunk 6]
```diff
@@ -89,36 +89,33 @@ internal fun CoroutineScope.attachForReadingDirectImpl(
...
-                        if (rc == 0) {
-                            channel.flush()
-                            selectable.interestOp(SelectInterest.READ, true)
-                            selector.select(selectable, SelectInterest.READ)
-                        }
-                    } while (rc == 0)
-                }
+        while (!channel.isClosedForWrite) {
+            timeout.withTimeout {
+                println("Read")
```
Comment (rsinukov) [inline ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt:95]: please remove `println` here and below

**Review by rsinukov — APPROVED** [review]

File: ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt (hunk 6 above)
Comment (e5l) [reply ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt]: fixed

**Pull request merged**